### PR TITLE
Add UUID Presto type

### PR DESCRIPTION
**Pull request by mbasmanova**
Summary: Add uuid() function and CAST to and from VARCHAR.

Differential Revision: D58222115

### CODE REVIEW

File: velox/functions/prestosql/tests/UuidFunctionsTest.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  std::unordered_set<std::string> uuids;
+  for (auto i = 0; i < 10; ++i) {
+    const auto uuid = result->valueAt(i).str();
+    ASSERT_EQ(36, uuid.size());
+    ASSERT_TRUE(uuids.insert(uuid).second);
+  }
+  ASSERT_EQ(10, uuids.size());
+}
+
+TEST_F(UuidFunctionsTest, castRoundTrip) {
```
Comment (amitkdutta) [inline velox/functions/prestosql/tests/UuidFunctionsTest.cpp:55]: We can add a test where casting from UUID to type other than varchar throws the exception.

File: velox/docs/develop/types.rst [hunk 2]
```diff
@@ -143,4 +144,4 @@ The high 52 bits of bigint store signed integer for milliseconds in UTC.
 Supported range of milliseconds is [0xFFF8000000000000L, 0x7FFFFFFFFFFFF]
 (or [-69387-04-22T03:45:14.752, 73326-09-11T20:14:45.247]). The low 12 bits
 store timezone ID. Supported range of timezone ID is [1, 1680].
-The definition of timezone IDs can be found in ``TimeZoneDatabase.cpp``.
\ No newline at end of file
+The definition of timezone IDs can be found in ``TimeZoneDatabase.cpp``.
```
Comment (amitkdutta) [inline velox/docs/develop/types.rst:147]: Related change?

**Review by amitkdutta — APPROVED** [review]
Looks good. Thanks @mbasmanova

File: velox/docs/develop/types.rst (hunk 2 above)
Comment (mbasmanova) [reply velox/docs/develop/types.rst:147]: There was no newline at the end of the file. My editor auto-added that.

File: velox/functions/prestosql/tests/UuidFunctionsTest.cpp (hunk 1 above)
Comment (mbasmanova) [reply velox/functions/prestosql/tests/UuidFunctionsTest.cpp:55]: Good idea. Adding

```
TEST_F(UuidFunctionsTest, unsupportedCast) {
  auto input = makeRowVector(ROW({}), 10);
  VELOX_ASSERT_THROW(
      evaluate("cast(uuid() as integer)", input),
      "Cannot cast UUID to INTEGER");
  VELOX_ASSERT_THROW(
      evaluate("cast(123 as uuid())", input), "Cannot cast BIGINT to UUID.");
}
```

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova thanks!

**Pull request closed without merging**